ballerina/common_utils.bal

```diff
@@ -83,6 +83,30 @@ isolated function getInvalidArgumentValueError(string listIndexOfError, string e
     }
 }
 
+isolated function getCycleRecursiveFragmentError(parser:FragmentNode spread, map<parser:FragmentNode> visitedSpreads)
+    returns ErrorDetail {
+    Location[] locations = [];
+    string spreadPath = "";
+    int listIndex = <int>visitedSpreads.keys().indexOf(spread.getName());
+    string[] spreadPathArray = visitedSpreads.keys().slice(listIndex);
+    if spreadPathArray.length() == 1 {
+        string message = string `Cannot spread fragment "${spread.getName()}" within itself.`;
+        return getErrorDetailRecord(message, <Location>spread.getSpreadLocation());
+    }
+    foreach int i in 1 ..< spreadPathArray.length() {
```

Comment:
What will happen if multiple recursive references are there inside the same fragment?

Eg.:

```graphql
{
    profile {
        ...PersonFragment
    }
}

fragment PersonFragment {
    ...Fragment1
    ...Fragment2
}

fragment Fragment1 {
    name
    ...PersonFragment
}

fragment Fragment2 {
    age
    ...PersonFragment
}
```

Shall we add such test as well?

Reply:
added a test